packages/lit-dev-content/site/docs/localization/runtime-mode.md

```diff
@@ -0,0 +1,308 @@
+---
+title: Runtime localization mode
+eleventyNavigation:
+  key: Runtime mode
+  parent: Localization
+  order: 2
+---
+
+In Lit Localize runtime mode, one JavaScript or TypeScript module is generated
+for each of your locales. Each generated module contains the localized templates
+for that locale. When your application switches locales, the module for that
+locale is imported, and all components are re-rendered.
+
+See [output modes](/docs/localization/overview/#output-modes) for a comparison
+of Lit Localize output modes.
+
+#### Example output
+
+```js
+// locales/es-419.ts
+export const templates = {
+  h3c44aff2d5f5ef6b: html`Hola <b>Mundo!</b>`,
+};
+```
+
+## Example of using runtime mode
+
+The following example demonstrates an application built with Lit Localize
+runtime mode:
+
+{% playground-example "docs/libraries/localization/runtime" "x-greeter.ts" %}
+
+The Lit GitHub repo includes full working examples
+([JavaScript](https://github.com/lit/lit/tree/main/packages/localize/examples/runtime-js),
+[TypeScript](https://github.com/lit/lit/tree/main/packages/localize/examples/runtime-ts))
+of Lit Localize runtime mode that you can use as templates.
+
+## Configuring runtime mode
+
+In your `lit-localize.json` config, set the `mode` property to `runtime`, and
+set the `output.outputDir` property to the location where you would like your
+localized template modules to be generated. See [runtime mode
+settings](/docs/localization/cli-and-config#runtime-mode-settings) for more
+details.
+
+In your JavaScript or TypeScript project, call `configureLocalization` function,
+passing an object with the following properties:
+
+- `sourceLocale: string`: Required locale code in which source templates in this
+  project are written, and the initial active locale.
+
+- `targetLocales: string[]`: Required locale codes that are supported by
+  this project. Should not include the `sourceLocale` code.
+
+- `loadLocale: (locale: string) => Promise<LocaleModule>`: Required function
+  that returns a promise of the localized templates for the given locale code.
+
+```js
+import {configureLocalization} from '@lit/localize';
+
+export const {getLocale, setLocale} = configureLocalization({
+  sourceLocale: 'en',
+  targetLocales: ['es-419', 'zh_CN'],
+  loadLocale: (locale) => import(`/locales/${locale}.js`),
+});
+```
+
+<div class="alert alert-info">
+
+Use the
+[`output.localeCodesModule`](/docs/localization/cli-and-config/#general-settings)
+setting to generate a module which exports a `sourceLocale` string and
+`targetLocales` array that can be passed to `configureLocalization`, to ensure
+that your runtime code remains in sync with your config file.
```

Comment:
I've rewritten the "Configuring runtime mode" section to  just directly walk through using this option, since while it's technically optional, I think everybody should do it.